### DISCUSSION TITLE
Option to compile epicscorelibs single-process for low memory machines?

### DISCUSSION THREAD

**Discussion by klenze** [dls-controls/pythonSoftIOC#127] [Q&A]
I tried to install softioc on a raspi3 (1GiB) with pip3 as recommended, but the setup system insists on using four processes in parallel, which will then slow down to a crawl when four g++ instances manage to eat all available memory. 

Is there a good way to tell pip/setuptools to only use a single thread, like make -j1 would?

I tried pip with --install-option="--jobs=1" but then it does not compile at all. 

Or would the recommended way be to compile EPICS manually and then have the softioc autodetect that installation?

Sorry if this question is off topic.

**Comment by AlexanderWells-diamond** [comment]
I don't think this is a PythonSoftIOC issue as such, hence why I've moved it to Discussions.

It may be possible to use a pre-compiled EPICS version, but PythonSoftIOC itself isn't what is compiling EPICS - we depend on [epicscorelibs](https://github.com/mdavidsaver/epicscorelibs) to do that for us. From what I understand of it, it itself uses [setuptools_dso](https://github.com/mdavidsaver/setuptools_dso) to build the shared objects for Python.

I don't know much beyond that unfortunately. The one suggestion I have is trying something like: `MAKEFLAGS="-j1" pip install softioc` . If it is using `make` anywhere down the chain, that may force it to use only a single process. Otherwise, installing `epicscorelibs` directly may provide other ways to configure the build process.

**Comment by klenze** [comment]
Unfortunately, make is not in the chain anywhere, and setuptools_dso's build_dso does not offer that option, it seems.

I have opened [an issue](https://github.com/mdavidsaver/epicscorelibs/issues/18) with epicscorelibs and hope they will be able to tell me how to do that. 

Thanks,
   Philipp
